tests/test_api.py

```diff
@@ -263,11 +284,13 @@ def test():
         t = Template("A{{ test().missingattribute }}B")
         pytest.raises(UndefinedError, t.render, test=test)
 
-    def test_undefined_and_special_attributes(self):
+    @staticmethod
+    def test_undefined_and_special_attributes():
         with pytest.raises(AttributeError):
             Undefined("Foo").__dict__
 
-    def test_undefined_attribute_error(self):
+    @staticmethod
+    def test_undefined_attribute_error():
```

Comment:
Just picking another of these changes randomly: I don't think there's any benefit in making methods static even if they don't use self as long as they usually get called through an instance of the class and not the class itself.